docs/cookbook/entities/custom-model-accessible-for-channel-admin.rst

```diff
@@ -0,0 +1,296 @@
+
+            $this->availableChannelsForAdminProvider = $availableChannelsForAdminProvider;
+            $this->channelRepository = $channelRepository;
+        }
+
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            /** @var ChannelInterface $adminChannel */
+            $adminChannel = $this->availableChannelsForAdminProvider->getChannel();
+
+            $builder
+                ->add('name', TextType::class, [
+                    'label' => 'Name'
+                ])
+                ->add('channel', ChannelChoiceType::class, [
+                    'choices' => $adminChannel === null ? $this->channelRepository->findAll() : [$adminChannel],
+                    'label' => 'sylius.ui.channel',
+                ])
+            ;
+        }
+
+        public function getBlockPrefix(): string
+        {
+            return 'supplier';
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Form\SupplierType:
+        arguments: ['@Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface']
+        tags: ['form.type']
+
+The ``Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface`` service allows getting a list of proper channels for the currently logged-in admin.
+
+Remember to register ``App\Form\SupplierType`` for resource:
+
+.. code-block:: yaml
+
+    sylius_resource:
+        resources:
+            app.supplier:
+                driver: doctrine/orm
+                classes:
+                    model: App\Entity\Supplier
+       +            form: App\Form\SupplierType
+
+
+1. Add your entity to admin menu:
+----------------------------------
+
+Adding a new entity to the admin menu is described in the section ``How to customize Admin Menu`` of :doc:`this guide </customization/menu>`.
+
+1. Restrict access to the entity for the respective channel administrator roles (using ACL/RBAC):
+-------------------------------------------------------------------------------------------------
+
+* Add access to all resource sections: ``index, create, update, delete, show, bulk_delete``
+
+.. code-block:: yaml
+
+    sylius_plus:
+        permissions:
+            app_admin_supplier_index:
+                parent: suppliers
+                label: action.index
+            app_admin_supplier_create:
+                parent: suppliers
+                label: action.create
+            app_admin_supplier_update:
+                parent: suppliers
+                label: action.update
+            app_admin_supplier_delete:
+                parent: suppliers
+                label: action.delete
+            app_admin_supplier_show:
+                parent: suppliers
+                label: action.show
+            app_admin_supplier_bulk_delete:
+                parent: suppliers
+                label: action.bulk_delete
+
+.. note::
+
+    More information about using administrator roles (ACL/RBAC) can be found :doc:`here </book/customers/admin_user>`.
```

Comment:
What will happen if we don't do that?

Reply:
It add actions to permissions tree

Reply:
so it is an RBAC feature. :) is it works without this config? I still think that it is a different feature, but if you really want to mention here about it, I Suggest moving it at the end with a note that it is additional stuff.